compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrVisitor.kt

```diff
@@ -692,6 +691,8 @@ class Fir2IrVisitor(
             is FirPropertyFromParameterResolvedNamedReference -> IrStatementOrigin.INITIALIZE_PROPERTY_FROM_PARAMETER
             is FirResolvedNamedReference -> when (resolvedSymbol) {
                 is AccessorSymbol, is SyntheticPropertySymbol -> IrStatementOrigin.GET_PROPERTY
+                is FirNamedFunctionSymbol -> if (resolvedSymbol.callableId.isInvoke()) IrStatementOrigin.INVOKE else null
+                is FirVariableSymbol<*> -> IrStatementOrigin.VARIABLE_AS_FUNCTION
```

Comment:
Why it's true if `statementOrigin` is called for `IrValueSymbol` (see line 720 below)?

Reply:
Not sure I understand the question. statementOrigin is called on FirReference, right?

Reply:
This function has two use-sites: one for `FirReference` of a function call (line 711), and another one for `FirReference` of a variable access (line 724). Your code is correct for a function call, but probably not correct for a variable access.

Reply:
Yeah, removed the case for FirVariableSymbol. The Invoke case seems ok, no matter which call sites.